pkg/util/common.go

```diff
@@ -1347,25 +1304,32 @@ func DeleteCustomResourceDefinition(apiExtensionClient *apiextensionsclient.Clie
 }
 
 // WaitUntilPodsAreReady will wait for the pods to be ready
-func WaitUntilPodsAreReady(clientset *kubernetes.Clientset, namespace string, labelSelector string, timeoutInSeconds int64) (bool, error) {
-	// timer starts the timer for timeoutInSeconds. If the task doesn't completed, return error
-	timeout := time.NewTimer(time.Duration(timeoutInSeconds) * time.Second)
-	// ticker starts and execute the task for every n intervals
-	ticker := time.NewTicker(10 * time.Second)
-	for {
-		select {
-		case <-timeout.C:
-			ticker.Stop()
-			return false, fmt.Errorf("[NS: %s | Label: %s] the pods weren't ready - timing out after %d seconds", namespace, labelSelector, timeoutInSeconds)
-		case <-ticker.C:
-			ready, err := IsPodReady(clientset, namespace, labelSelector)
-			if err != nil || ready {
-				timeout.Stop()
-				ticker.Stop()
-				return ready, err
-			}
+func WaitUntilPodsAreReady(clientset *kubernetes.Clientset, namespace string, labelSelector string, timeoutInSeconds int64) error {
+	numOfTimesToCheck := 10
+	var duration time.Duration
+	if int(timeoutInSeconds) < numOfTimesToCheck {
```

Comment:
This logic isn't correct.  If you set duration to timeoutInSeconds, then numOfTimesToCheck needs to be set to 1.  Otherwise you are checking 10*timeout.

Reply:
I can fix that, it was a feature, not a bug 😄, but I can change the feature.

Reply:
That's a confusing feature then. :) . Timeout should be the total time waited in any condition.  Making it wait 10*timeout is not at all behavior I would expect as a user.